sdk/monitor/azquery/custom_client.go

```diff
@@ -109,47 +111,82 @@ func (e *ErrorInfo) Error() string {
 // Row of data in a table, types of data used by service specified in LogsColumnType
 type Row []any
 
-// Table - Contains the columns and rows for one table in a query response.
-type Table struct {
-	// REQUIRED; The list of columns in this table.
-	Columns []*Column `json:"columns,omitempty"`
+// TimeInterval specifies the time range over which to query.
+// Use NewTimeInterval() for help formatting.
+// Follows the ISO8601 time interval standard with most common format being startISOTime/endISOTime.
+// Use UTC for all times.
+type TimeInterval string
 
-	// REQUIRED; The name of the table.
-	Name *string `json:"name,omitempty"`
+// NewTimeInterval creates a ISO8601TimeInterval for use in a query.
+// Use UTC for start and end times.
+func NewTimeInterval(start time.Time, end time.Time) TimeInterval {
+	return TimeInterval(start.Format(time.RFC3339) + "/" + end.Format(time.RFC3339))
+}
 
-	// REQUIRED; The resulting rows from this query.
-	Rows []Row `json:"rows,omitempty"`
+// Times returns the interval's start and end times if it's in the format startISOTime/endISOTime, else it will return an error.
+func (i TimeInterval) Times() (time.Time, time.Time, error) {
+	// split into different start and end times
+	times := strings.Split(string(i), "/")
+	if len(times) != 2 {
+		return time.Time{}, time.Time{}, errors.New("time interval should be in format startISOTime/endISOTime")
+	}
+	start, err := time.Parse(time.RFC3339, times[0])
+	if err != nil {
+		return time.Time{}, time.Time{}, errors.New("error parsing start time")
+	}
+	end, err := time.Parse(time.RFC3339, times[1])
+	if err != nil {
+		return time.Time{}, time.Time{}, errors.New("error parsing end time")
+	}
+	// return times
+	return start, end, nil
+}
 
-	// maps column name to index for easy lookup, helper for accessing Row data
-	ColumnIndexLookup map[string]int `json:"-"`
+// LogsQueryOptions sets server timeout, query statistics and visualization information
+type LogsQueryOptions struct {
+	// Set Statistics to true to get logs query execution statistics,
+	// such as CPU and memory consumption. Defaults to false.
+	Statistics *bool
+
+	// Set Visualization to true to get visualization
+	// data for logs queries. Defaults to false.
+	Visualization *bool
+
+	// By default, the Azure Monitor Query service will run your
+	// query for up to three minutes. To increase the default timeout,
+	// set Wait to desired number of seconds.
+	// Max wait time the service will allow is ten minutes (600 seconds).
+	Wait *int
 }
 
-// UnmarshalJSON implements the json.Unmarshaller interface for type Table.
-func (t *Table) UnmarshalJSON(data []byte) error {
-	var rawMsg map[string]json.RawMessage
-	if err := json.Unmarshal(data, &rawMsg); err != nil {
-		return fmt.Errorf("unmarshalling type %T: %v", t, err)
+// String implements a custom string method for type LogsQueryOptions.
```

Comment:
Nit: "...implements the fmt.Stringer interface for..."

Reply:
fixed in commit 
f54dc06